js/mailuniverse.js

```diff
@@ -391,14 +387,27 @@ function MailUniverse(callAfterBigBang, online, testOptions) {
           console.warn('OF EMAILS, maybe some PASSWORDS.  This was turned on');
           console.warn('via the secret debug mode UI.  Use it to turn us off:');
           console.warn('https://wiki.mozilla.org/Gaia/Email/SecretDebugMode');
-          $log.DEBUG_realtimeLogEverything(dump);
-          slog.setSensitiveDataLoggingEnabled(true);
+          logic.realtimeLogEverything();
+          logic.setEventCensor(null);
         }
         else if (self.config.debugLogging !== 'dangerous') {
           console.warn('GENERAL LOGGING ENABLED!');
           console.warn('(CIRCULAR EVENT LOGGING WITH NON-SENSITIVE DATA)');
-          $log.enableGeneralLogging();
-          slog.setSensitiveDataLoggingEnabled(false);
+          // Recursively redact any references to mail headers and any detail
+          // properties starting with an underscore.
+          function censorObject(x) {
+            for (var key in x) {
+              var value = x[key];
+              if (key[0] === '_' || (value && (value.suid || value.srvid))) {
```

Comment:
What's the rationale on sanitizing suid and srvid?  suid is entirely generated by us and I've found it to potentially be handy to correlate what I see in a log with a bug filer's STR's or videos.  Specifically to find the log part that lines up with the alleged problem.  srvid is server issued, but should be a similarly opaque value.  One handy thing from not censoring it is it potentially would help us help technical users diagnose a problem without them having to provide the uncensored log.  On the flip side, I've never done this, so that's an argument against srvid I guess.

I don't really care too much either way, especially since right now I think I'm depending on console.log stuff rather than slog/logic stuff, but comments for rationale are appropriate here.